subprojects/core/src/main/java/org/gradle/api/internal/DynamicModulesClassPathProvider.java

```diff
@@ -123,6 +123,7 @@ private ClassPath plusExtensionModules(ClassPath classpath, Set<Module> extensio
     };
 
     private static final String[] GRADLE_OPTIONAL_EXTENSION_MODULES = {
+        "gradle-daemon-services",
```

Comment:
Why are these marked optional?

Reply:
That's the magic incantation to make all of the tests pass. This PR is just about getting the buckets into place. We can solve all the weird problems in the various tests later.